src/transformers/pipelines/text2text_generation.py

```diff
@@ -15,49 +15,52 @@
 
 
 @add_end_docstrings(PIPELINE_INIT_ARGS)
-class SummarizationPipeline(Pipeline):
+class Text2TextGenerationPipeline(Pipeline):
     """
-    Summarize news articles and other documents.
+    Pipeline for text to text generation using seq2seq models.
 
-    This summarizing pipeline can currently be loaded from :func:`~transformers.pipeline` using the following task
-    identifier: :obj:`"summarization"`.
+    This Text2TextGenerationPipeline pipeline can currently be loaded from :func:`~transformers.pipeline` using the
+    following task identifier: :obj:`"text2text-generation"`.
 
-    The models that this pipeline can use are models that have been fine-tuned on a summarization task, which is
-    currently, '`bart-large-cnn`', '`t5-small`', '`t5-base`', '`t5-large`', '`t5-3b`', '`t5-11b`'. See the up-to-date
-    list of available models on `huggingface.co/models <https://huggingface.co/models?filter=summarization>`__.
+    The models that this pipeline can use are models that have been fine-tuned on a translation task. See the
+    up-to-date list of available models on `huggingface.co/models <https://huggingface.co/models?filter=seq2seq>`__.
 
     Usage::
 
-        # use bart in pytorch
-        summarizer = pipeline("summarization")
-        summarizer("Sam Shleifer writes the best docstring examples in the whole world.", min_length=5, max_length=20)
-
-        # use t5 in tf
-        summarizer = pipeline("summarization", model="t5-base", tokenizer="t5-base", framework="tf")
-        summarizer("Sam Shleifer writes the best docstring examples in the whole world.", min_length=5, max_length=20)
+        text2text_generator = pipeline("text2text-generation")
+        text2text_generator("question: What is 42 ? context: 42 is the answer to life, the universe and everything")
     """
 
+    return_name = "generated"
```

Comment:
what is this?

Reply:
Is it solely used to name the outputs correctly? Maybe add a nice comment to explain its usage so we know what its purpose is in the future